woocommerce/woocommerce
# Monorepo Utils: Add changelog addition from actions

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Adds an action to call the [CLI changefile addition tool](https://github.com/woocommerce/woocommerce/pull/38357).

The CLI will be called on pull request creation, added commits, and description editing. A template edition that adds fields for authors to fill out is also included. When the box is ticked, the automation will add changefile entries to all affected projects based on the data supplied in the description.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fork this repo and push this branch (`add/add-changelog-action`) to your fork.
2. Merge this branch into `trunk`.
3. Create a new branch from `trunk` called `test/add/add-changelog-action` and make a few changes to projects with no changelogger, ie `tools/`. Commit and push changes to your fork. Create a pull request into your fork's `trunk` branch. The pull request will now include a section to enter changelog details. Leave this section unchanged.
4. See the `Changelog Auto Add` PR check kick off. Click to view. Confirm the job succeeds and results in a message similar to `PR #71 does not have the "Automatically create a changelog entry from the details" checkbox checked. No changelog will be created.`
5. Now edit the PR description and fill out the changelog details. The `Changelog Auto Add` will kick off again once you save the description. See the check succeed once again with a message like `No projects require a changelog`.
6. Now make a change in a project that does have changelogger, like `woocommerce-admin`. Once changes are committed and pushed up, see the `Changelog Auto Add` check kickoff once again. This time, a changefile will be added based on the data provided in the PR description.
7. Confirm the changefile create exists and reflects the data in the description.
8. Change the description, save, and see the job run again. All changefiles will reflect the change.
9. Now undo a change in a project. See the changefile removed.
